### DISCUSSION TITLE
Add Hs to a reaction

### DISCUSSION THREAD

**Discussion by thorlucas** [rdkit/rdkit#3612] [Q&A] [answered]
Hi,

I have a SMARTS reaction that is a full reaction, i.e. it's not a pattern. I'd like to call AddHs on it so that when I draw this reaction, it draws it with the Hs (rather than putting a dot for the lone electrons).

AddHs only takes a molecule.

How do I do this?

**Comment by greglandrum** [comment, marked as answer]
If the only reason you want to add the Hs is to have the molecule drawn without radicals, it's easier to just set the `includeRadicals` flag to False:
![image](https://user-images.githubusercontent.com/540511/101176692-1fdbab80-3647-11eb-8852-3f2603efce9c.png)
Notice that I'm also using the `fromSmiles` argument to `ReactionFromSmarts()`

Actually adding Hs is more complicated and you need to come in from SMARTS instead of SMILES:
![image](https://user-images.githubusercontent.com/540511/101177504-2ae30b80-3648-11eb-8554-5dc30acd57e1.png)
also notice that I've made the bond orders explicit in the input SMARTS

If you actually want to use SMILES as the input things get considerably more complicated. I'm not going to show that here because I'm not convinced it's actually useful.

**Comment by thorlucas** [comment]
Thanks!
